[Tan Hin Khai Stephen] iP

blackonyyx opened the pull request:
1. Duke Prototype Works
2. Add Text UI Script Testing
3. Add Proposed commands: todo, deadline, event
4. Add saving and loading from text files
5. Add DukeException Classes
6. Enumerated Duke Exception Technical Messages
7. Enumerated commands to duke
8. Add main loop elements to UI class.
9. Add jar file
10. Add datetime parsing and autocorrection (Not fully functional)

Commit:
Add Gradle support

blackonyyx committed:
Level 1 added

blackonyyx committed:
Level 2 is here

blackonyyx committed:
Level 3

blackonyyx committed:
A-Classes used classes to represent tasks

blackonyyx committed:
Added Inheritance for tasks and managed tasks via manager

blackonyyx committed:
Added sh script automated testing

blackonyyx committed:
Added Enums, Packaged Related classes together, Error handling

blackonyyx committed:
Added Delete command to the list of supported commands

blackonyyx committed:
Added IO file handling to the Chatbot Application such that the Chatbot
can now save the tasklist to a textfile.
Also added tag for packages that was already done in a previous
iteration of the project.

blackonyyx committed:
Added datetime parsing to the TimedTask Classes
1. Added datetime functionalities to the timed task
2. Added datetime difference to response message by Duke
3. Added String cleaning strip() to Duke inputs.
4. Added new Help Messages
5. Added Autocorrect function using time of initial execution as default
datetime.
6. Added Deadlines and Event occurance countdowns from given datetime
parsed to the current date.

blackonyyx committed:
Merge branch 'branch-Level-8'
Merge Branch 7 and Branch 8 into master

blackonyyx committed:
More OOP
1. Refactor Duke.java and ChatboxApplication.java with Mainloop in
ChatboxApplication
    This is in order to perform future UI implementations and allow
    interfacing between Chatbot back end and front end.
2. Add Text Parser class
    This is in order to perform text string parsing for user inputs.

blackonyyx committed:
1. Add jUnit Testing on Timed Task and TextParser Class
    Tested for Expected Exceptions and Expected returns from methods
    called
    This is due to these classes being most essential for proper
    functioning of the main loop and to handle the user input.

blackonyyx committed:
Branch for Level 9 of Duke Project.
1. Add find function
    Find function performs a search in the description of each task
    This is done using Regex to find matching patterns, rather than just
    using CharSequences.

blackonyyx committed:
Make Code adheres to java code standards of module.

To allow for better code review and understanding by other peer
reviewers

blackonyyx committed:
Add javadocs for all constructors and class headers,
as well as javadocs for most methods inside the classes

blackonyyx committed:
Merge branch 'A-JavaDoc' of https://github.com/blackonyyx/ip
Added Checkstyle and JavaDocs into IP

blackonyyx committed:
Merge branch 'branch-Level-9' of https://github.com/blackonyyx/ip

blackonyyx committed:
Merge branch 'add-gradle-support' of https://github.com/blackonyyx/ip

blackonyyx committed:
Add gradle integration into master branch.
This allows for more automated integration of testing and compilation
for the production version.
Use gradle to refactor and update more documentation for javadocs
This allowed better updating of documentations

rtshkmr commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,143 @@
+            case BYE:
+                return Boolean.TRUE.toString();
+            case HELP:
+                return this.help();
```
Hi! Really love the implementation of a HELP case

rtshkmr commented on src/main/java/exceptions/DukeCommandException.java:
```diff
@@ -0,0 +1,14 @@
+     * Constructor class for DukeCommandException
+     * @param bad_cmd the command that is unrecognisable by Duke Application
+     */
+    public DukeCommandException(String bad_cmd) {
```
I think we should camelCase our identifiers, so `bad_cmd` should be `badCmd`. snake_case is reserved for test method naming!

rtshkmr commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,48 @@
+ * 5. Blank Command Given: For handling when a Blank command is given to a input.
+ * 6. Index Error: When a invalid index is given
+ * 7. UnknownException: For handling anything exceptionally unexpected
+ */
```
Really love the comprehensive documentation for your class. Learnt something new from this, good job man :+1:

rtshkmr commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,48 @@
+public abstract class DukeException extends Exception {
+
+    String bad_cmd;
+    ErrorEncode code;
```
I think it's a good idea to always put access modifiers to your fields, for comprehensiveness-sake.

rtshkmr commented on src/main/java/exceptions/DukeException.java:
```diff
@@ -0,0 +1,48 @@
+    ErrorEncode code;
+
+    /**
+     * Constructor class for a Generic DukeException for any errortype encountered in Duke
```
JDoc guidelines suggest we should phrase is like so :
`Constructs....`

You might have missed it out for your constructors, but this week's CheckStyle requirement probably means that you already resolved such issues so this might be a moot point I'm making

rtshkmr commented on src/main/java/exceptions/ErrorEncode.java:
```diff
@@ -0,0 +1,49 @@
+package exceptions;
+
+public enum ErrorEncode {
```
Really like your use of Enums to consolidate your Errors in one place, did something similar for mine too!

petrickjerico replied:
Agree with the previous reviewer! Grouping all your Errors in one Enum class makes the code neater.

On another note, noticed you added a header comment for the Command Enum class. For consistency, perhaps one for this class too?

rtshkmr commented on src/main/java/tasks/TimedTask.java:
```diff
@@ -0,0 +1,91 @@
+    static {
+        DateTimeFormatterBuilder dateTimeFormatterBuilder = new DateTimeFormatterBuilder();
+        dateTimeFormatterBuilder.appendPattern("dd-MM-yyyy");
+        dateTimeFormatterBuilder.parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0);
```
The use of ChronoField is new to me, thanks for showing this alternative way to handle Date/Time :+1:

blackonyyx replied:
Sorry the review was done a while ago and i din respond, but thanks for the review! Glad to know you learnt something!

rtshkmr reviewed (commented):
Hi, your code is really nice to read, it's clean and naming is aptly done. Flows really well. Couldn't nitpick on any specific naming violations though. Good job man! :+1:

blackonyyx committed:
Enforced CheckStyle on all violating code blocks
This will make code more readable and comply with the code standards

petrickjerico commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,66 @@
+
+    /**
+     * Checklist icon for displaying in the toString Representation.
+      * @return either done or not done
```
Found a hard catch here: misaligned asterisk. _Sneaky_.

petrickjerico commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    /**
+     * Mark a generic Task object as done
```
Here, maybe start with `Marks...` instead of `Mark...`? 

I noticed the same issue in several other lines too, but as the previous reviewer mentioned, this comment could prove redundant.

petrickjerico commented on src/main/java/tasks/TextParser.java:
```diff
@@ -0,0 +1,54 @@
+     * @param cmd the string command for the Duke Application
+     * @return Command Enumeration
+     */
+
```
JDoc seems to be against any "blank line between the documentation block and the method/class." Perhaps I can suggest removing the (unintentional?) blank at line 38 here.

Kindly survey the other lines, as I might have missed some other cases of this!

petrickjerico commented on src/main/java/tasks/TaskManager.java:
```diff
@@ -0,0 +1,224 @@
+     * @throws DukeIoException If no loaded save is read
+     */
+    public TaskManager(String path) throws DukeIoException {
+        this.ioParser = new TaskIoParser(path);
+        this.taskList = ioParser.loadTaskList();
+        this.textParser = new TextParser();
```
Referring to the JDoc for all rules (the one suggested for W3 was only basic + intermediate), the `this` keyword is best used "only when a field is shadowed by a method or constructor parameter".

In this case, since `path` does not shadow any of the `TaskManager` attributes (`taskList`, `ioParser`, and `textParser`), perhaps it is safe and good practice to remove the `this` keyword.

blackonyyx replied:
:0 oh noes im used to the python way of using this/self at all times. Thanks for alerting me to this. Sorry for the late response to your review!

petrickjerico reviewed (commented):
Hi, blackonyyx! After reviewing your PR, I found that your code overall looks very neat and clean 👍🏼 I found some library usages that stood as insightful and worth trying out on my own too. The code could use some minor cosmetics, but otherwise, the refactoring is thorough and effective. Kudos!

blackonyyx committed:
Add a GUI for the Duke Application
Add JavaFX for duke application todo: edit the linebreaks and wrappers
to get a better display.
TODO: Add initial message and username setting

blackonyyx committed:
Add Major overhaul and refactor of code base to follow a more functional
style.

Refactor of Commands inside TaskList to the CommandList Library
This will prevent TaskManager Class from being overbulked with methods,
and modularises command tasks to a functional interface.

Refactor of User Interface into a Interface class that is to be used by both
GUI and CLI methods of execution.

blackonyyx committed:
Add new refactoring of GUI to have better feel.
Refactor of GUI to use GUIHelper class to handle the interface between
the backend and front end of script.

Proper indentation of most tasks have been performed.

Help Command can now support command based help searches.

blackonyyx committed:
Amend Code to conform with Checkstyle requirements.

blackonyyx committed:
Minor String Formatting modifications

blackonyyx committed:
Merge pull request #1 from blackonyyx/branch-A-Assertions

Minor String Formatting modifications

blackonyyx committed:
Minor Code Formatting modifications

blackonyyx committed:
Merge branch 'master' into branch-A-CodeQuality

blackonyyx committed:
Create gradle.yml

blackonyyx committed:
Merge pull request #4 from blackonyyx/Continuous-Integ-2

Create gradle.yml 
Performed Continuous Integration

blackonyyx committed:
Merge pull request #3 from blackonyyx/branch-A-CodeQuality

Minor Code Formatting modifications

blackonyyx committed:
stuff

blackonyyx committed:
Merge branch 'master' of https://github.com/blackonyyx/ip

blackonyyx committed:
Amend Code to contain less this.<attribute / method> to conform with the
code standards of module.

blackonyyx committed:
Add User Guide to the project repository and update README.md to reflect
the state of the project as being completed.

User Guide will provide a good reference for enduser to understand how
to execute commands and tasks using Duke.

blackonyyx committed:
Add README.md and Userguide help.

blackonyyx committed:
Set theme jekyll-theme-dinky

blackonyyx committed:
Merge branch 'master' of https://github.com/blackonyyx/ip

blackonyyx committed:
Add README.md and Userguide help.

blackonyyx committed:
Set theme jekyll-theme-dinky

blackonyyx committed:
Removed boiler plate code

blackonyyx committed:
Merge branch 'master' of https://github.com/blackonyyx/ip
